R/Factbase.R

```diff
@@ -342,6 +345,20 @@ value_type_for_vector <- function(v, column_name) {
         stop(paste('Cannot work out which data type to use for column', column_name, 'containing a', typeof(v), 'vector.  Only Date, POSIXt, text or real are accepted'))
 }
 
+ensureDefinitionHyperlinkOwnerSupplied <- function(definition, hyperlink, owner) {
+    if (is.null(definition))
+        stop("argument \"definition\" is missing")
+    if (!is.character(definition) || length(definition) != 1)
+        stop("argument \"definition\" must be a character vector of length 1, or null")
+    if (is.null(hyperlink))
+        stop("argument \"hyperlink\" is missing")
+    if (!is.character(hyperlink) || length(hyperlink) != 1)
+        stop("argument \"hyperlink\" must be a character vector of length 1")
+    if (is.null(owner))
+        stop("argument \"owner\" is missing")
+    if (!is.character(owner) || length(owner) != 1)
+        stop("argument \"owner\" must be a character vector of length 1")
+}
```

Comment:
FYI: I think it is easier to consolidate these conditions into the simpler `stopifnot` function when the `stop` message isn't too long and the condition isn't too long.

```
stopifnot("argument \"owner\" must be a character vector of length 1" = is.character(owner) && length(owner) == 1L)
```

Reply:
I tried this out and I found the code harder to read, so I'm leaving it as is.